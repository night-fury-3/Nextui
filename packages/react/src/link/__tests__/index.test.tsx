import React from "react";
import {mount} from "enzyme";

import Link from "../index";

describe("Link", () => {
  it("should render correctly", () => {
    const wrapper = mount(
      <div>
        <Link href="https://v1.nextui.org">link</Link>
        <Link color="secondary" href="https://v1.nextui.org">
          link
        </Link>
        <Link isExternal color="success" href="https://v1.nextui.org">
          link
        </Link>
        <Link underline href="https://v1.nextui.org">
          link
        </Link>
        <Link block href="https://v1.nextui.org">
          link
        </Link>
      </div>,
    );

    expect(() => wrapper.unmount()).not.toThrow();
  });

  it("should be no errors when href missing", () => {
    const errorSpy = jest.spyOn(console, "error");
    const wrapper = mount(<Link />);

    expect(errorSpy).not.toHaveBeenCalled();
    expect(() => wrapper.unmount()).not.toThrow();
    errorSpy.mockRestore();
  });

  it("should forward ref", () => {
    let ref = React.createRef<HTMLAnchorElement>();
    const errorSpy = jest.spyOn(console, "error");
    const wrapper = mount(<Link ref={ref} />);

    expect(errorSpy).not.toHaveBeenCalled();
    expect(ref.current).not.toBeNull();
    expect(() => wrapper.unmount()).not.toThrow();
    errorSpy.mockRestore();
  });

  it("an warning should be thrown when using the pure prop", () => {
    const errorSpy = jest.spyOn(console, "error").mockImplementation(() => {});

    mount(<Link pure />);
    expect(errorSpy).toHaveBeenCalled();
    errorSpy.mockRestore();
  });
});
