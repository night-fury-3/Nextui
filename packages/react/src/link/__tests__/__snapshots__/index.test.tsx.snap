// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should render correctly 1`] = `"<div><a href=\\"https://nextui.org\\" class=\\"nextui-c-gUrcXm nextui-c-gUrcXm-eAdznm-color-default nextui-c-gUrcXm-iFdQKc-animated-true\\">link</a><a href=\\"https://nextui.org\\" class=\\"nextui-c-gUrcXm nextui-c-gUrcXm-krgohU-color-secondary nextui-c-gUrcXm-iFdQKc-animated-true\\">link</a><a href=\\"https://nextui.org\\" class=\\"nextui-c-gUrcXm nextui-c-gUrcXm-byQUqd-color-success nextui-c-gUrcXm-iFdQKc-animated-true\\">link<svg class=\\"nextui-c-hdKjck nextui-link-icon\\" fill=\\"none\\" height=\\"1em\\" shape-rendering=\\"geometricPrecision\\" stroke=\\"currentColor\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"1.5\\" viewBox=\\"0 0 24 24\\" width=\\"1em\\"><path d=\\"M18 13v6a2 2 0 01-2 2H5a2 2 0 01-2-2V8a2 2 0 012-2h6\\"></path><path d=\\"M15 3h6v6\\"></path><path d=\\"M10 14L21 3\\"></path></svg></a><a href=\\"https://nextui.org\\" class=\\"nextui-c-gUrcXm nextui-c-gUrcXm-eAdznm-color-default nextui-c-gUrcXm-cSuHME-underline-true nextui-c-gUrcXm-iFdQKc-animated-true\\">link</a><a href=\\"https://nextui.org\\" class=\\"nextui-c-gUrcXm nextui-c-gUrcXm-eAdznm-color-default nextui-c-gUrcXm-eIcaMf-block-true nextui-c-gUrcXm-iFdQKc-animated-true nextui-c-gUrcXm-bhXeQK-cv\\">link</a></div>"`;
